test/hotspot/jtreg/serviceability/sa/jmap-hprof/JMapHProfLargeHeapTest.java

```diff
@@ -82,15 +82,22 @@ private static void testHProfFileFormat(String vmArgs, long heapSize,
         procBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
         Process largeHeapProc = procBuilder.start();
 
-        try (Scanner largeHeapScanner = new Scanner(
-                largeHeapProc.getInputStream());) {
+        try (BufferedReader r = new BufferedReader(
+                new InputStreamReader(largeHeapProc.getInputStream()))) {
             String pidstring = null;
-            if (!largeHeapScanner.hasNext()) {
-                throw new RuntimeException ("Test failed: could not open largeHeapScanner.");
+            while ((pidstring = r.readLine()) != null) {
+                // The output might contain different VM output, skip it while searching PID line.
+                if (pidstring.matches("PID\\[[0-9].*\\]")) {
+                    System.out.println("Found: " + pidstring);
+                    break;
+                } else {
+                    System.out.println("Ignoring: " + pidstring);
+                }
```

Comment:
Do we really want this in the test output?

Reply:
It might be helpful to ensure that we don't mess with matching PID line if test fails.